library/nxos_zone_zoneset.py

```diff
@@ -79,6 +79,7 @@
                                 description:
                                     - devtype of the zone member used along with Smart zoning config
                                 choices: ['initiator', 'target', 'both']
+                                required: true
```

Comment:
very good point. i just updated the code again. 

I tried to use required_if in the arg spec to require devtype when smart_zoning was true, but because of how the nested parameters were working even when devtype is not defined in the play it defaults it to null. this always passed the required_if validation so i had to fail the module manually when the devtype check takes place. i also set smart_zoning to required as well. 

This seems to work well and provide pretty specific error reporting. Let me know what you think.